packages/smithy-client/src/client.ts

```diff
@@ -1,11 +1,19 @@
 import { constructStack } from "@smithy/middleware-stack";
-import { Client as IClient, Command, MetadataBearer, MiddlewareStack, RequestHandler } from "@smithy/types";
+import {
+  Client as IClient,
+  Command,
+  FetchHttpHandlerOptions,
+  MetadataBearer,
+  MiddlewareStack,
+  NodeHttpHandlerOptions,
+  RequestHandler,
+} from "@smithy/types";
 
 /**
  * @public
  */
 export interface SmithyConfiguration<HandlerOptions> {
-  requestHandler: RequestHandler<any, any, HandlerOptions>;
+  requestHandler: RequestHandler<any, any, HandlerOptions> | NodeHttpHandlerOptions | FetchHttpHandlerOptions | any;
```

Comment:
the input type of `requestHandler` on `SmithyConfiguration` is loosened to allow for ctor-param-pass-through, and then below, the type of `SmithyResolvedConfiguration` is retained as it was.

Reply:
Is loosening the type to `any` ok?

Reply:
changed to Record<string, unknown>